src/components/AlbulList.js

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import IconButton from '@mui/material/IconButton';
+import CloseIcon from '@mui/icons-material/Close';
+import Snackbar from '@mui/material/Snackbar';
+import {
+  Container,
+  TextField,
+  Button,
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  Typography,
+  Box,
+  Grid,
+} from '@mui/material';
+
+const AlbumsList = () => {
+  const [albums, setAlbums] = useState([]); // for albums
+  const [limit, setLimit] = useState(1); // for pagination
+  const [searchType, setSearchType] = useState('albumName'); //default value for select input
+  const [searchTerm, setSearchTerm] = useState(''); // for search input filed value
+  const [message, setMessage] = useState(''); // for not found message
+  const [hasError, setHasError] = useState(false); //for error during fetching from API
+
+  //make an API call with search values to backend and return the result
+  const fetchAlbums = async (searchType, searchTerm, limit) => {
+    try {
+      const response = await axios.get(
+        'http://localhost:8000/api/v1/albums/filter',
+        {
+          params: {
+            limit,
+            [searchType]: searchTerm,
+          },
+        }
+      );
+      const searchResult = response.data.albums;
+      setAlbums(searchResult);
+      if (searchResult.length === 0) {
+        setMessage('No result found');
+      }
+    } catch (error) {
+      console.error('Error fetching albums:', error);
+      setHasError(true);
+    }
+  };
+
+  //call the function to make API request for search input value
+  const handleSearch = () => {
+    setLimit(1);
+    fetchAlbums(searchType, searchTerm, 1);
```

Comment:
Why are we doing `setLimit(1)` here? it hasn't changed right?

And we can pass in the limit in the `fetchAlbums` call instead of hardcoding it so that if we ever _do_ change the limit there's not too many places to edit:

```suggestion
    setLimit(1);
    fetchAlbums(searchType, searchTerm, limit);
```

Reply:
Updated it